packages/runtime-common/realm.ts

```diff
@@ -1943,78 +1932,23 @@ export class Realm {
     return data;
   }
 
-  private listeningClients: WritableStream[] = [];
-
-  private async subscribe(
-    request: Request,
-    requestContext: RequestContext,
-  ): Promise<Response> {
-    let headers = {
-      'Content-Type': 'text/event-stream',
-      'Cache-Control': 'no-cache',
-      Connection: 'keep-alive',
-    };
-
-    let { response, writable } = this.#adapter.createStreamingResponse(
-      request,
-      requestContext,
-      {
-        status: 200,
-        headers,
-      },
-      () => {
-        this.listeningClients = this.listeningClients.filter(
-          (w) => w !== writable,
-        );
-        if (this.listeningClients.length === 0) {
-          this.#adapter.unsubscribe();
-        }
-      },
-    );
-
-    // To remove as part of CS-8104, a workaround for a file watcher irregularity in tests
-    let subscribeOptions: { watcher?: string } = {};
-
-    let testFileWatcher = new URL(request.url).searchParams.get(
-      'testFileWatcher',
-    );
-
-    if (testFileWatcher) {
-      subscribeOptions.watcher = testFileWatcher;
-    }
-
-    if (this.listeningClients.length === 0) {
-      await this.#adapter.subscribe((data) => {
-        let tracked = this.getTrackedWrite(data);
-        if (!tracked || tracked.isTracked) {
-          return;
-        }
-        this.broadcastRealmEvent(data);
-        this.#updateItems.push({
-          operation: ('added' in data
-            ? 'add'
-            : 'updated' in data
-              ? 'update'
-              : 'removed') as UpdateItem['operation'],
-          url: tracked.url,
-        });
-        this.drainUpdates();
-      }, subscribeOptions);
-    }
-
-    this.listeningClients.push(writable);
-
-    if (testFileWatcher) {
-      this.sendServerEvent({
-        type: 'test-file-watcher-ready',
+  private async startFileWatcher() {
```

Comment:
@backspace  is there any reason why this should be async? usually the pattern is that a synchronous method calls an async "drain" method...

Reply:
`start()` is async though?

Reply:
everything in this method is being called synchronously, no? I don't see any awaits in your function. why should be be async?

Reply:
eep that’s actually a bug, it should be `await this.#adapter.subscribe(…`

Reply:
The `Realm.start()` is async because it's awaiting other things--but not this.

Reply:
ah, so there is async in here, then we are just missing the await